File: Creational-Pattern/builder-pattern/src/Java/Builder-Pattern/src/com/manhpd/Employee.java
package com.manhpd;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@AllArgsConstructor
@NoArgsConstructor
public class Employee {

	private int id;

	private String firstName;

	private String lastName;

	private String position;

	public Employee(EmployeeBuilder builder) {
		this.id = builder.id;
		this.firstName = builder.firstName;
		this.lastName = builder.lastName;
		this.position = builder.position;
	}

	@Override
	public String toString() {
		return "Id: " + id + "\nFirst name: " + firstName + "\nLast name: " + lastName + "\nPosition: " + position;
	}

	public static class EmployeeBuilder {

		private int id;

		private String firstName;

		private String lastName;

		private String position;

		public EmployeeBuilder id(int id) {
			this.id = id;
			return this;
		}

		public EmployeeBuilder firstName(String firstName) {
			this.firstName = firstName;
			return this;
		}

		public EmployeeBuilder lastName(String lastName) {
			this.lastName = lastName;
			return this;
		}

		public EmployeeBuilder position(String position) {
			this.position = position;
			return this;
		}

		public Employee build() {
			return new Employee(this);
		}
	}

}

File: Creational-Pattern/builder-pattern/src/Javascript/Employee.js

class Employee {

    constructor(builder) {
        this._id = builder.id;
        this._firstName = builder.firstName;
        this._lastName = builder.lastName;
        this._position = builder.position;
    }

    get id() {
        return this._id;
    }

    get firstname() {
        return this._firstName;
    }

    get lastName() {
        return this._lastName;
    }

    get position() {
        return this._position;
    }

    static get EmployeeBuilder() {
        class EmployeeBuilder {
            constructor(id) {
                this._id = id;
            }

            withFirstName(firstName) {
                this._firstName = firstName;
                return this;
            }

            withLastName(lastName) {
                this._lastName = lastName;
                return this;
            }

            withPosition(position) {
                this._position = position;
                return this;
            }

            build() {
                return new Employee(this);
            }
        }

        return EmployeeBuilder;
    }
}